Docs/rust_build.md

```diff
@@ -0,0 +1,115 @@
+
+3. Verify `cargo` is working:
+
+   \>`cargo --version`
+
+4. Install the desired Rust tool chain
+
+   - Example: `1.68.2 x86_64 toolchain`
+
+   - Windows:
+
+      \>`rustup toolchain install 1.68.2-x86_64-pc-windows-msvc`
+
+      \>`rustup component add rust-src --toolchain 1.68.2-x86_64-pc-windows-msvc`
+
+   - Linux:
+
+      \>`rustup toolchain install 1.68.2-x86_64-unknown-linux-gnu`
+
+      \>`rustup component add rust-src --toolchain 1.68.2-x86_64-unknown-linux-gnu`
+
+5. Install `cargo make`
+
+   \>`cargo install --force cargo-make`
+
+At this point, the essential Rust applications are installed, and a repo can begin to add and build Rust code.
+
+## Build Variations
+
+First, the simplest case is a new user approaching a package that supports Rust modules. In that case, just build the
+package like "normal" (e.g., stuart build) and the "normal" build output (i.e., EFI drivers and firmware volumes) will
+be produced.
+
+### Build an Individual Module
+
+  ```cmd
+  cargo make build <Module Name>
+  ```
+
+The following command line options are available:
+
+  1. `-p PROFILE [development|release]`. `DEFAULT` = `development (debug)`
+  2. `-e ARCH=[IA32|X64|LOCAL]`. `DEFAULT` = `X64`
+  3. `-e TARGET_TRIPLE=[triple]`.
+
+  - `ARCH=LOCAL` is used to build any locally executable tools associated with a Rust library package (e.g., a
+  dual-purpose executable and library).
+
+  - `TARGET_TRIPLE=<triple>` is used to cross-compile locally executable tools associated with a Rust library package.
+
+  - Note: A Rust package must be specified.
+  - The output location is:
+    - `target/[x86_64-unknown-uefi|i686-unknown-uefi]/[debug|release]/module_name.[efi|rlib]`
+
+### Test an Individual Module
+
+  ```cmd
+  cargo make test <Optional: Module Name>
+  ```
+
+  - Note: If a package is not specified, all packages will be tested.
+
+### Supported Build Combinations
+
+1. C source + Rust source mixed in INF (Library or Module)
+   - Rust source code is supported by build rule – `Rust-To-Lib-File` (`.rs` => `.lib`)
+
+   - Limitation: Rust code cannot have external dependencies.
+
+2. Pure Rust Module only.
+
+   - A `Cargo.toml` file is added to the INF file `[Sources]` section.
+
+   - Rust Module build is supported by build rule – `Toml-File.RUST_MODULE` (`Toml` => `.efi`)
+
+   - Limitation: Runtime might be a problem, such as virtual address translation for Rust code internal global
+     variables.
+
+3. Pure Rust Module + Pure Rust Library with Cargo Dependency.
+
+   - Cargo dependency means the Rust lib dependency declared in `Cargo.toml`.
+
+4. Pure Rust Module + C Library with EDK II Dependency.
+
+   - Rust Module build is supported by build rule – `Toml-File` (`Toml` => `.lib`)
+
+   - The EDK II dependency means the EDK II lib dependency declared in INF.
+
+     - If a Rust module is built with C, the cargo must use `staticlib`. Otherwise, `rlib` is used.

```

Comment:
I think this should be a bit more exact. i.e. specifically call out (or link to rust docs https://doc.rust-lang.org/cargo/reference/cargo-targets.html#library) that they need to set crate-type = ['staticlib'] at the minimum under the [lib] section.

Reply:
Added